src/main/java/agh/ics/oop/SimulationEngine.java

```diff
@@ -1,25 +1,29 @@
 package agh.ics.oop;
-
+import java.util.List;
 public class SimulationEngine implements IEngine {
     private MoveDirection[] moves;
     private IWorldMap map;
     public SimulationEngine(MoveDirection[] moves,IWorldMap map, Vector2d [] positions)
     {
         this.moves = moves;
         this.map = map;
+
         for(Vector2d x : positions )
         {
-            map.place(new Animal(map,x));
+            Animal animal =  new Animal(map,x);
+            map.place(animal);
+            animal.addObserver((IPositionChangeObserver) map);
```

Comment:
Jeśli każda mapa rozszerzałaby ten interfejs to nie musielibyśmy rzutować -> interfejsy też mogą rozszerzać inne interfejsy.